### PR TITLE
Problem bug report

### PR DESCRIPTION
**Pull request by geniusgordon**
tag form add bootstrap
back to problem panel in edit problem
judge type add prefix

### CODE REVIEW

File: problem/templates/problem/problem.html [hunk 1]
```diff
@@ -2,10 +2,10 @@
 <div class="pname_block panel">
   <div class="left">
     <div class="pname">
-      <h2>{{ problem.pname }} &nbsp;&nbsp;
+      <h2>problem #{{ problem.pk }}:<br>{{ problem.pname }} &nbsp;&nbsp;
```
Comment (nickelcheng) [inline problem/templates/problem/problem.html]: ```{{ problem.pk }} - ```{{ problem.pname }}}``` is enough

File: problem/templates/problem/edit.html [hunk 2]
```diff
@@ -9,7 +9,12 @@
 {% endblock %}
 {% block body_block %}
 <div class="container" id="Body">
-  <h4>You are editing problem {{pid}}: {{pname}}</h4>
+  <h4>
+    You are editing problem {{pid}}: {{pname}}
```
Comment (nickelcheng) [inline problem/templates/problem/edit.html]: ```{{ pid }} - {{ pname }}``` is enough

Comment (nickelcheng) [inline problem/templates/problem/edit.html]: Maybe you can use a larger font size

**Comment by nickelcheng** [comment]
1. You need to lengthen max length of judge type
default
2. Please add another button go back to problem (just like what contest did)
default
3. Please customize html title for each problem detail / edit page (add pid)

**Comment by nickelcheng** [comment]
Please fix this comment: "為什麼只有 Problem 頁面的 EDIT 是藍色字加底線 其他都是按鈕呢 ? 排擠 EDIT 嗎 ?"

**Comment by geniusgordon** [comment]
fixed

**Comment by nickelcheng** [comment]
Please also add problem id edit problem title.

**Comment by nickelcheng** [comment]
+1 after the comment fixed

**Comment by nickelcheng** [comment]
1. Please also add problem id edit problem title.
2. Other judge id can not be empty when judge source is other judge
3. I got this again XD
screenshot from 2015-04-28 15 45 38
I choose "Use Other Judge" and leave judge type empty, then push save button.
I think the judge type column should not be empty by default, and you should block invalid choices in forms.py (you can ask @henryyang42)

**Comment by geniusgordon** [comment]
fixed

**Comment by nickelcheng** [comment]
+1, @bruce3557 please check

File: problem/static/problem/js/edit.js [hunk 3]
```diff
@@ -200,18 +203,14 @@ function choose_judge_type(option) {
 function choose_judge_source(option) {
     if (option == "OTHER") {
         $("#id_other_judge_id").parent().parent().show();
-	$("#id_judge_type option").hide();
-	$("option[value='UVA']").show();
-	$("option[value='POJ']").show();
-	$("option[value='ICPC']").show();
-	$("#id_judge_type").val("UVA");
+    	$("option[value^='OTHER_']").show();
+        $("option[value^='LOCAL_']").hide();
```
Comment (bruce3557) [inline problem/static/problem/js/edit.js]: Please refine alignment between lines

File: problem/static/problem/js/edit.js [hunk 4]
```diff
@@ -200,18 +203,14 @@ function choose_judge_type(option) {
...
     } else if (option == "LOCAL") {
 	    console.log("local");
-	$("#id_judge_type option").show();
-	$("option[value='UVA']").hide();
-	$("option[value='POJ']").hide();
-	$("option[value='ICPC']").hide();
-	$("#id_judge_type").val("NORMAL");
+    	$("option[value^='OTHER_']").hide();
+        $("option[value^='LOCAL_']").show();
+        $("#id_judge_type").val($("option[value^='LOCAL_']")[0].value);
```
Comment (bruce3557) [inline problem/static/problem/js/edit.js]: Ditto

**Comment by bruce3557** [comment]
+1 besides the comments

**Comment by geniusgordon** [comment]
fixed

**Comment by nickelcheng** [comment]
going to merge.

**Pull request merged**